core/src/main/java/bio/terra/pearl/core/service/export/EnrolleeImportService.java

```diff
@@ -346,7 +346,7 @@ protected SurveyResponse importSurveyResponse(UUID portalId, SurveyFormatter for
 
             List<ParticipantTask> tasks = surveyTaskDispatcher.assign(assignDto, studyEnv.getId(),
                     new ResponsibleEntity(DataAuditInfo.systemProcessName(getClass(), "handleSurveyPublished.assignToExistingEnrollees")));
-            relatedTask = tasks.get(0);
+            relatedTask = tasks.stream().filter(task -> task.getEnrolleeId().equals(enrollee.getId())).findFirst().orElse(null);
```

Comment:
Instead of doing this, I think you want to change line 343 above from `null` to `List.of(enrollee.getId())` -- that will limit the assignment to the current enrollee